Sync SIPNET user/model docs with current implementation (#380)
PecanProject/sipnet: docs/model-structure.md

Reviewer: ```suggestion
- Nitrogen (when enabled): leaf-on reallocation changes plant N storage by $\Delta N = C_{\text{leafon}}(1/CN_{\text{leaf}} - 1/CN_{\text{wood}})$. Positive $\Delta N$ is drawn from storage and can limit leaf-on; negative $\Delta N$ is returned to storage.

The carbon constraint, and the nitrogen constraint when applicable, scale the transfer down. The scaled amount is applied to the pools and is not subsequently reduced by the general nitrogen limitation (Sec. [Nitrogen Limitation](#nitrogen-limitation)).
```

Reviewer: ```suggestion
from the leaf carbon pool to the litter pool (or soil pool, if the litter pool is not being used). When `nitrogen-cycle` is enabled, the nitrogen corresponding to the transferred leaf carbon follows it, except for the fraction resorbed to the plant nitrogen storage pool. When `nitrogen-cycle` is disabled, no nitrogen transfer or resorption is calculated.
```

### Leaf On/Leaf Off

Leaf on and leaf off events define the timing of leaf emergence and senescence, respectively. These events do not carry

> their own magnitude parameters; instead, the transferred carbon and nitrogen are computed from model state and the

Reviewer: revising for clarity

```suggestion
Leaf on and leaf off events define the timing of leaf emergence and senescence, respectively. These events only specify the date, not the amount, of leaf biomass C and N added or subtracted; instead, the transferred carbon and nitrogen are computed from model state and the
```

> parameters `leafGrowth`, `leafOnReallocFrac`, `fracLeafFall`, and `leafNResorptionFrac`.

Reviewer: @copilot Please use mathematical symbols consistently in this section. Before replacing the source code names here, add leafOnReallocFrac and leafNResorptionFrac, with symbols that follow conventions in that file, to docs/parameters.md - they are currently absent. Then use those symbols throughout lines 1214–1235, not only in this paragraph.

Please inform me if there are other places where source code names are used in this file and should be replaced.
